fe/fe-core/src/main/java/org/apache/doris/fs/remote/S3FileSystem.java

```diff
@@ -20,36 +20,36 @@
 import org.apache.doris.analysis.StorageBackend;
 import org.apache.doris.backup.Status;
 import org.apache.doris.common.UserException;
-import org.apache.doris.common.security.authentication.AuthenticationConfig;
 import org.apache.doris.common.security.authentication.HadoopAuthenticator;
-import org.apache.doris.datasource.property.PropertyConverter;
-import org.apache.doris.datasource.property.constants.S3Properties;
+import org.apache.doris.common.util.S3URI;
+import org.apache.doris.datasource.property.storage.AbstractS3CompatibleProperties;
 import org.apache.doris.fs.obj.S3ObjStorage;
-import org.apache.doris.fs.remote.dfs.DFSFileSystem;
 
-import com.amazonaws.services.s3.model.AmazonS3Exception;
 import com.google.common.annotations.VisibleForTesting;
-import com.google.common.base.Strings;
-import org.apache.hadoop.conf.Configuration;
-import org.apache.hadoop.fs.FileStatus;
 import org.apache.hadoop.fs.FileSystem;
-import org.apache.hadoop.fs.Path;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import software.amazon.awssdk.services.s3.S3Client;
 
-import java.io.FileNotFoundException;
-import java.io.IOException;
+import java.util.HashSet;
 import java.util.List;
-import java.util.Map;
+import java.util.Set;
 
 public class S3FileSystem extends ObjFileSystem {
 
     private static final Logger LOG = LogManager.getLogger(S3FileSystem.class);
     private HadoopAuthenticator authenticator = null;
+    private AbstractS3CompatibleProperties s3Properties;
 
-    public S3FileSystem(Map<String, String> properties) {
-        super(StorageBackend.StorageType.S3.name(), StorageBackend.StorageType.S3, new S3ObjStorage(properties));
+
+    public S3FileSystem(AbstractS3CompatibleProperties s3Properties) {
+
+        super(StorageBackend.StorageType.S3.name(), StorageBackend.StorageType.S3,
+                new S3ObjStorage(s3Properties));
+        this.s3Properties = s3Properties;
```

Comment:
Why need 2 properties? `s3Properties` and `storageProperties`?

Reply:
Each file system has its own parameter class with specific methods. Using the concrete type improves clarity and avoids unnecessary abstraction. 

Since StorageProperties was somewhat redundant, I refactored it into an abstract base class. This allows each subclass to simply return its own concrete instance while still sharing a common structure.